taxi/tests/test_views_car.py

```diff
@@ -0,0 +1,183 @@
+            license_number="XXX12345"
+        )
+
+        self.client.force_login(self.user)
+        self.list_response = self.client.get(CARS_URL)
+        self.create_response = self.client.get(
+            reverse("taxi:car-create"))
+        self.update_response = self.client.get(reverse(
+            "taxi:car-update", args=[1]))
+        self.delete_response = self.client.get(reverse(
+            "taxi:car-delete", args=[1]))
+
+    def test_car_views_url_exists_at_desired_location(self):
+        list_view_response = self.client.get("/cars/")
+        create_view_response = self.client.get("/cars/create/")
+        update_view_response = self.client.get("/cars/3/update/")
+        delete_view_response = self.client.get("/cars/6/delete/")
+
+        self.assertEqual(list_view_response.status_code, 200)
+        self.assertEqual(create_view_response.status_code, 200)
+        self.assertEqual(update_view_response.status_code, 200)
+        self.assertEqual(delete_view_response.status_code, 200)
+
+    def test_car_views_url_accessible_by_name(self):
+        self.assertEqual(self.list_response.status_code, 200)
+        self.assertEqual(self.create_response.status_code, 200)
+        self.assertEqual(self.update_response.status_code, 200)
+        self.assertEqual(self.delete_response.status_code, 200)
+
+    def test_car_views_uses_correct_template(self):
+        self.assertTemplateUsed(self.list_response, "taxi/car_list.html")
+        self.assertTemplateUsed(self.create_response, "taxi/car_form.html")
+        self.assertTemplateUsed(self.update_response, "taxi/car_form.html")
+        self.assertTemplateUsed(self.delete_response,
+                                "taxi/car_confirm_delete.html")
+
+    def test_car_list_view_pagination_is_five(self):
+        self.assertTrue("is_paginated" in self.list_response.context)
+        self.assertTrue(self.list_response.context["is_paginated"] is True)
+        self.assertEqual(
+            len(self.list_response.context["car_list"]), 5)
+
+    def test_car_list_view_lists_all_cars(self):
+        response_sec_page = self.client.get(CARS_URL + "?page=2")
+        cars = Car.objects.all()
+
+        self.assertEqual(
+            list(
+                self.list_response.context["car_list"]
+            ) + list(
+                response_sec_page.context["car_list"]
+            ), list(cars))
+
+    def test_car_list_view_search(self):
+        manufacturer = Manufacturer.objects.create(
+            name="Opel",
+            country="Germany",
+        )
+        number_of_cars = 7
+        for car_id in range(number_of_cars):
+            Car.objects.create(
+                model=f"Astra {car_id}",
+                manufacturer=manufacturer,
+            )
+        response_list = []
+        for page_number in range(1, 3):
+            search_result = self.client.get(
+                CARS_URL + "?model=Model" + f"&page={page_number}",
+            )
```

Comment:
No need to fix it for now, but for future:
![image](https://user-images.githubusercontent.com/44442087/216758350-81f819af-63e6-4854-9091-411474b1ffb2.png)
This can be used to pass query parameters instead of changing the url